src/formElementPlugins/dropdownChoice/src/ApplicationView.tsx

```diff
@@ -1,47 +1,67 @@
-import React, { useEffect } from 'react'
+import React, { useEffect, useState } from 'react'
 import { Dropdown, Header, Label } from 'semantic-ui-react'
 import { ApplicationViewProps } from '../../types'
 
 const ApplicationView: React.FC<ApplicationViewProps> = ({
   parameters,
   onUpdate,
-  value,
-  setValue,
+  // value,
+  // setValue,
   isEditable,
   currentResponse,
   validationState,
   onSave,
   Markdown,
   getDefaultIndex,
 }) => {
-  const { label, description, placeholder, search, options, default: defaultOption } = parameters
+  const {
+    label,
+    description,
+    placeholder,
+    search,
+    options,
+    optionsDisplayProperty,
+    default: defaultOption,
+  } = parameters
+
+  const [selectedIndex, setSelectedIndex] = useState<number>()
 
   useEffect(() => {
-    onUpdate(value)
     // This ensures that, if a default is specified, it gets saved on first load
-    if (!value && defaultOption !== undefined) {
-      const defaultIndex = getDefaultIndex(defaultOption, options)
-      const defaultValue = options?.[defaultIndex]
+    if (!currentResponse?.text && defaultOption !== undefined) {
+      const optionIndex = getDefaultIndex(defaultOption, options)
       onSave({
-        text: defaultValue,
-        optionIndex: defaultIndex,
+        text: optionsDisplayProperty
+          ? options[optionIndex][optionsDisplayProperty]
+          : options[optionIndex],
```

Comment:
Same here

Reply:
I'm getting the feeling I might be missing what you're referring to. Just to clarify:
- if `optionsDisplayProperty` is not defined, it'll assume the options is an array of text values
- if `optionsDisplayProperty` IS defined, it'll display that field from each object in the array
- if `optionsDisplayProperty` is defined but that field doesn't exist, it'll throw an error and not display the plugin

Is this not what we want (Or is it not actually doing what I think it's doing?)

Reply:
That's ok to throw the error for 3rd options "if `optionsDisplayProperty` is defined but that field doesn't exist, it'll throw an error and not display the plugin", I was just checking what would happen in this case.